src/main/java/com/example/demo/model/ReservationRequestModel.java

```diff
@@ -0,0 +1,93 @@
+package com.example.demo.model;
+
+import com.example.demo.core.Pair;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class ReservationRequestModel {
+
+
+    private Long movieId; 
+    private String name;
+    private String surname;
+    // (seatId,ticketType)
+    private List<Pair<Long, String>> seats;
+
+    ReservationRequestModel() {
+    }
+
+    public ReservationRequestModel(Long movieId, String name, String surname, List<Pair<Long, String>> seats) {
+        this.movieId = movieId;
+        this.name = name;
+        this.surname = surname;
+        this.seats = seats;
+    }
+
+    public boolean validIdTicketTypes() {
+        List<String> allowed = new ArrayList<>(List.of("adult", "student", "child"));
+
+        for (Pair<Long, String> seat : seats) {
+            if (!allowed.contains(seat.getSecond())) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public Double toPrice() {
```

Comment:
$$ w double to nie najlepszy pomysł

Reply:
zgodzę się ze dokładniejszym rozwiązaniem byłaby Cena  przechowywana jako liczba całkowita groszy